### ISSUE TITLE
saliency maps for all DL models

### ISSUE THREAD

**Issue by krrome** [kipoi/kipoi#224]
- [x] gradient functions for Keras models
 - [x] gradient functions for pytorch models
 - [x] gradient functions for TensorFlow models

API:

 - [x] centralised way of accessing gradient functions through API
 - [x] allow report of gradients only
 - [x] allow report of grad*input
 - [x] visualisation for grad*input: detect DNA sequences automatically and use seqlogos for those.

**Comment by krrome** [comment]
fixed in #261
